src/zip_eq_impl.rs

```diff
@@ -52,6 +52,25 @@ where
         }
     }
 
+    fn fold<B, F>(self, init: B, mut f: F) -> B
+    where
+        Self: Sized,
+        F: FnMut(B, Self::Item) -> B,
+    {
+        let Self { a, mut b } = self;
+
+        let acc = a.fold(init, |acc, x| match b.next() {
+            Some(y) => f(acc, (x, y)),
+            None => panic!("itertools: .zip_eq() reached end of one iterator before the other"),
```

Comment:
You could `b.next().expect(..)`. I doubt it's faster though.

Reply:
just tested it, no difference